epochtalk/epochtalk
# Permissions upgrade

Test Procedure
1. Delete all `roles` from the database, ensure that custom and base permissions are populated in the database after server restart.
2. Delete all `base_permissions` from the `roles` table, ensure that custom and base permissions are populated in the database after server restart.
3. Delete all `custom_permissions` from the `roles` table, ensure that custom and base permissions are populated in the database after server restart.
4. Change the value of a module permission, if the role has the default value for that permission, the value should update upon server restart. If the role modified the default value for that permission, already, the value shouldn't change.
5. Delete module permissions, the custom permissions should be updated to reflect this change.
6. Add module permissions, the custom permissions should be updated to reflect this change.
7. Mix changes to module permissions, combine steps 4-6 and check that the custom permissions reflect this change on server restart.

Note: custom roles, will not be updated automatically, If new permissions are added these will have to be manually added via the roles management page. New permissions will default to false for custom roles.
